File: backend/tmp_cleanup.py
import sqlite3
import os

db_path = "backend/data/autostonks.db"
if not os.path.exists(db_path):
    print(f"Database not found at {db_path}")
    exit(1)

conn = sqlite3.connect(db_path)
cursor = conn.cursor()

targets = [
    "asddasdasdasdasd",
    "$asddasdasdasdasd",
    "So like what do u want",
    "asd",
    "aw"
]

for target in targets:
    cursor.execute("DELETE FROM strategies WHERE name = ?", (target,))
    print(f"Deleted {cursor.rowcount} strategy records matching name: '{target}'")
    
    # Delete from strategy_positions
    cursor.execute("DELETE FROM strategy_positions WHERE strategy_id NOT IN (SELECT id FROM strategies)")
    if cursor.rowcount > 0:
        print(f"Cleaned up {cursor.rowcount} orphaned positions.")

    # Delete from trade_log
    cursor.execute("DELETE FROM trade_log WHERE strategy_id NOT IN (SELECT id FROM strategies)")
    if cursor.rowcount > 0:
        print(f"Cleaned up {cursor.rowcount} orphaned trade logs.")

conn.commit()
conn.close()
print("Cleanup complete!")

File: backend/tmp_cleanup.rs
use rusqlite::Connection;
use std::path::Path;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let db_path = "data/autostonks.db";
    if !Path::new(db_path).exists() {
        println!("Database not found at {}", db_path);
        return Ok(());
    }

    let conn = Connection::open(db_path)?;
    
    let targets = vec![
        "asddasdasdasdasd",
        "$asddasdasdasdasd",
        "So like what do u want",
        "asd",
        "aw"
    ];

    for target in targets {
        let deleted = conn.execute(
            "DELETE FROM strategies WHERE name = ?1",
            [target],
        )?;
        println!("Deleted {} strategy records matching name: '{}'", deleted, target);
        
        // Also delete from trade_log if any
        let trade_deleted = conn.execute(
            "DELETE FROM trade_log WHERE strategy_id IN (SELECT id FROM strategies WHERE name = ?1)",
            [target],
        )?;
        if trade_deleted > 0 {
            println!("Deleted {} trade log records for strategy: '{}'", trade_deleted, target);
        }
    }

    println!("Cleanup complete!");
    Ok(())
}
